mlflow/server/js/src/experiment-tracking/components/ExperimentPage.js

```diff
@@ -95,31 +95,59 @@ export class ExperimentPage extends Component {
     this.detectNewRunsTimer = setInterval(() => this.detectNewRuns(), DETECT_NEW_RUNS_INTERVAL);
   }
 
-  componentDidUpdate(prevProps) {
-    this.maybeReloadData(prevProps);
+  componentDidUpdate(prevProps, prevState) {
+    this.maybeReloadData(prevProps, prevState);
+  }
+
+  /** Snapshots desired attributes of the component's current state in local storage. */
+  snapshotComponentState() {
+    const store = ExperimentPage.getLocalStore(this.props.experimentId);
+    store.saveComponentState(new ExperimentPagePersistedState(this.state.persistedState));
   }
 
   static getDerivedStateFromProps(props, state) {
+    const store = ExperimentPage.getLocalStore(props.experimentId);
     if (props.experimentId !== state.lastExperimentId) {
       return {
         persistedState:
           state.lastExperimentId === undefined
             ? state.persistedState
-            : new ExperimentPagePersistedState({ orderByKey: DEFAULT_ORDER_BY_KEY }).toJSON(),
+            : new ExperimentPagePersistedState({
+                ...store.loadComponentState(),
+              }).toJSON(),
         lastExperimentId: props.experimentId,
-        lifecycleFilter: LIFECYCLE_FILTER.ACTIVE,
         nextPageToken: null,
         getExperimentRequestId: getUUID(),
         searchRunsRequestId: getUUID(),
         ...PAGINATION_DEFAULT_STATE,
       };
+    } else if (
+      // react to url being changed by user or using the back button
+      props.location.search !== state.urlState &&
```

Comment:
Maybe we can just merge the return for the common set of values? Basically dedup the return and the common fields that need to reset based on the URL change or the experimentId change which is also part of the URL

Reply:
Good point, I'll look into this